CIP-0XXX/README.md

```diff
@@ -0,0 +1,540 @@
+Category: ?
+Status: Proposed
+Authors:
+    - Jean-Philippe Raynaud <jp.raynaud@gmail.com>
+    - Arnaud Bailly <arnaud.bailly@iohk.io>
+    - Marcin Szamotulski <marcin.szamotulski@iohk.io>
+    - Armando Santos <armando.santos@iohk.io>
+    - Neil Davies <neil.davies@iohk.io>
+    - Sebastian Nagel <sebastian.nagel@ncoding.at>
+    - ?
+Implementors: 
+    - Cardano Scaling team <https://github.com/cardano-scaling>
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/?
+Created: 2024-08-02
+License: Apache-2.0
+---
+
+# Cardano Decentralized Message Queue
+
+## Abstract
+
+We propose to create a decentralized message diffusion protocol leveraging the Cardano network layer. This protocol allows to follow a topic based diffusion of messages from publishers to subscribers in a decentralized way.  
+
+The messages can be sent and received by nodes running in a non intrusive way side by side to the Cardano node in order to enable inter-nodes communications.
+
+In this way, we can significantly reduce the cost and effort required to build a decentralised network for message diffusion by using Cardano's established infrastructure, with limited impact on the performance and no impact on the security of the Cardano network.
+
+# Motivation
+
+Mithril is a protocol based on a [Stake-based Threshold Multi-signature scheme](https://iohk.io/en/research/library/papers/mithril-stake-based-threshold-multisignatures/) which leverages the Cardano SPOs to certify data from the Cardano chain in a trustless way. Mithril is currently used in the Cardano ecosystem in order to enable fast bootstrapping of full nodes and enabling secure light wallets.
+
+The Mithril protocol coordinates the collection of individual signatures originating from the signers (run by SPOs) by the aggregators which combine them into Mithril multi-signatures and certificates. In order to be fully decentralized, the protocol needs to rely on a decentralized peer to peer network which, if built from the ground up, would require significant efforts and investment. Furthermore, the majority of SPO's, as the representatives of Cardano's active stake, will have to adopt and operate Mithril nodes alongside their Cardano node. Thus a natural solution is to use the Cardano network layer to significantly facilitate the development of Mithril protocol without a significant impact on the Cardano network or high maintenance efforts for the SPOs. 
+
+Other protocols in the Cardano ecosystem, such as Leios and Peras (and probably other protocols in the future), also need the capability to diffuse messages originating from block producers in a decentralized fashion. However, in the Leios and Peras cases, the Cardano node itself is a producer and consumer of these messages. We have taken into consideration this need for a generic solution in the design proposed. 
+
+The proposed solution is described in detail below.
+
+# Specification
+
+## Overview
+
+![Overview](./img/overview.jpg)
```

Comment:
Notably, this diagram does _not_ say that "network node" would be the cardano node. It could be any node implementing this protocol, even a "mithril network node" as you have sketched it out in that "# Information Diffusion Architecture" section.

Reply:
I think that should be made more clear in the accompanying text then.  Some people will expect that it's `cardano-node`.

Reply:
@ch1bo since we're proposing that `Message Producers` are communicating directly with themselves, that should be indicated in the diagram.  Only showing `n2n` communication of `cardano-node` suggests that will be used for message dessimination.

Reply:
@coot the `Message Producers` will be, in the case of Mithril, the existing `Mithril signers` which will send signatures with the  `n2c` mini-protocols. 

In any case, I will update the document to explain what is meant by `Network Node`, `Message Producer` and `Message Consumer` in general and in particular for Mithril (referring to the `# Information Diffusion Architecture` section) :+1:

Reply:
Done in https://github.com/cardano-scaling/CIPs/commit/1796de29ea0adc201a98a532d013723fec66a879